src/main/java/com/gmail/yuramitryahin/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,42 @@
+package com.gmail.yuramitryahin.web.filter;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/driver/registration")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null) {
```

Comment:
what if logged in user deletes themselves?

Reply:
in my task, the user cannot delete himself, but if user was deleted he cannot log in

Reply:
"In AuthenticationFilter, think if this check driverService.get(userId) == null will actually work.
If inside your driverService.get() you are currently calling get() on Optional, you can omit this check in AuthenticationFilter"